qlib/finco/utils.py

```diff
@@ -3,14 +3,16 @@
 from fuzzywuzzy import fuzz
 
 
-class Singleton:
+class Singleton(type):
     _instance = None
 
-    def __new__(cls, *args, **kwargs):
+    def __call__(cls, *args, **kwargs):
         if cls._instance is None:
-            cls._instance = super().__new__(cls, *args, **kwargs)
+            cls._instance = super(Singleton, cls).__call__(*args, **kwargs)  
         return cls._instance
 
+class SingletonBaseClass(metaclass=Singleton):  
```

Comment:
I tried [this example](https://github.com/you-n-g/cheatsheets/blob/master/language/python/meta.py#L74)  before.
I can't understand why we need two level of class for Singleton?